### PR TITLE
Fix CoGroupByKey by Converting to Bytes After Join

### PR DESCRIPTION
**Pull request by jasonfic**

### CODE REVIEW

File: af2_dags/dependencies/dataflow_scripts/dataflow_utils/dataflow_utils.py
Comment (woodjt) [inline af2_dags/dependencies/dataflow_scripts/dataflow_utils/dataflow_utils.py:1]: Is there a reason to create a copy of datum with the changed vals (converted_datum) rather than change directly in datum. is this an immutability related concept? this is fine, and we can easily revert, just not sure of the logic in having essentially 2 copies of datum at the end (one converted and one not)

**Review by woodjt — APPROVED** [review]

File: af2_dags/dependencies/dataflow_scripts/dataflow_utils/dataflow_utils.py
Comment (jasonfic) [reply af2_dags/dependencies/dataflow_scripts/dataflow_utils/dataflow_utils.py:1]: The main reason I have the second copy of datum was for when you want to convert the names of the datum keys to byte strings in addition to the values (e.g., b'parc_num', b'18C00002348'). If I didn't have that copy, we would need to insert a new key to the original datum and delete the non-converted key afterwards (e.g., datum[str(key).encode('utf-8') -> del datum[key]). This saves an operation.
